course/functions.php

```diff
@@ -96,4 +98,49 @@ function course_register_block_patterns_category() {
 	);
 }
 
-add_action( 'init', 'course_register_block_patterns_category' );
\ No newline at end of file
+add_action( 'init', 'course_register_block_patterns_category' );
+
+/**
+ * Determine the theme variation and save in option.
+ *
+ * @param int     $post_id Post ID.
+ * @param WP_Post $post    Post object.
+ * @param bool    $update  Whether this is an existing post being updated or not.
+ */
+function course_save_global_styles( $post_id, $post, $update ) {
+	if ( 'wp_global_styles' !== $post->post_type ) {
+		return;
+	}
+
+	$global_styles      = json_decode( $post->post_content, true );
+	$global['settings'] = isset( $global_styles['settings'] ) ? $global_styles['settings'] : array();
+	$global['styles']   = isset( $global_styles['styles'] ) ? $global_styles['styles'] : array();
+	$variations         = WP_Theme_JSON_Resolver::get_style_variations();
+	$current_variation  = 'default';
+	foreach ( $variations as $variation ) {
+		if ( $variation['settings'] === $global['settings'] && $variation['styles'] === $global['styles'] ) {
+			$current_variation = strtolower( $variation['title'] );
```

Comment:
What do you think of using the `sanitize_title` function instead of `strtolower`? Just in case we start using titles with multiple words.